Openshift template

Jiri-Kremser opened the pull request:
https://issues.jboss.org/browse/HAWKULAR-1182

(note to myself) current issues:
* Running Cassandra as root user or group is not recommended - please start Cassandra using a different system user.
* http://pastebin.com/zVKSwhS4


workaround: `oc adm policy add-scc-to-user anyuid -z default`


@pilhuhn could you please look? I used a similar ansible playbook as Juca in [here](https://github.com/hawkular/hawkular-apm/blob/master/examples/vertx-opentracing/vertx-opentracing.yml).

todo:
* [x] the `openshift/origin-metrics-cassandra:v1.4.1` image use the default path for storing the data, but we need to store it to a mounted volume to make the data durable. Currently, it fails because of access rights.

pilhuhn commented on openshift/hawkular-services-template.yaml:
```diff
@@ -0,0 +1,199 @@
+    name: hawkular-cassandra
+  spec:
+    ports:
+    - name: thrift
```
I think this is no longer needed with Inventory on Postgres

pilhuhn commented on openshift/hawkular-services-ansible-vars.yaml:
```diff
@@ -0,0 +1,34 @@
+
+hawkular_services_image: pilhuhn/hawkular-services:0.30.0.Final
+
+cassandra_image: cassandra:3.7
```
H-services currently uses 3.0.9 -- we may revisit that though.

pilhuhn commented:
Both images need a physical volume claim 
H-services for /opt/data
And Cassandra for /var/lib or whatever the image exposes

Jiri-Kremser commented:
> Both images need a physical volume claim

@pilhuhn problem with this is that developer can't create the PVs, I can create 2 PVC in this template but it will not start, because the admin has to add those to PVs. I can do that in that ansible script as system:admin, but it's getting complicated

theute commented:
Is template-without-pvs.yaml the right name ? I guess it means "template without the PV definitions" but I first thought it was "teamplate that doesn't need pvs" and then I couldn't understand the difference with -ephemeral.
It could be just me...

theute commented:
I would appreciate if the route name could be passed in parameter rather than hardcoded to public (I would prefer "hawkular-services-myproject..."

Jiri-Kremser commented:
@theute:
> I would appreciate if the route name could be passed in parameter rather than

done

> Is template-without-pvs.yaml the right name ? I guess it means

renamed

I've added a condition to find out what version of `oc` user is running and using the right syntax for it. I've tested both templates on versions:

* v1.3.2
* v1.4.1
* v1.5.0-alpha2

And it worked for all. Yesterday, I came home where I have different network and run the `./startEphemeral.sh` again and h-services were not able to resolve the `hawkular-cassandra` hostname. Flushing the iptables didn't help, but what helped in my case was `systemctl stop firewalld.service rpcbind.service rpcbind.socket`. I've put that in the readme just for sure.

If there is nothing else ( @pilhuhn ? ), I think the PR is now good to go.

Jiri-Kremser commented:
retest this please

pilhuhn commented:
The "This PR potentially fails with hawkular-client-ruby." is actually expected.

pilhuhn commented on openshift/startEphemeral.sh:
```diff
@@ -0,0 +1,82 @@
+OC_CLUSTER_VERSION="v1.4.1"
+
+prepare_cluster(){
+  oc cluster up --version=$OC_CLUSTER_VERSION && \
```
If oc cluster up fails, the new project will not be created ?

Jiri-Kremser replied:
nope (`&&`)
